ElDorito/Source/Patches/ContentItems.cpp

```diff
@@ -248,24 +266,23 @@ namespace
 		closedir(dp);
 	}
 
+	void DelAllBLFContentItems()
+	{
+		//reset array indices to 0 so we can repopulate the content items from scratch (see sub_0x55B410)
```

Comment:
This is insufficient.

`contentItemsGlobal` should be a pointer to a `Blam::DataArray<c_content_item>`, but most of the code in this file was written before we had the `DataArray` type defined. There are more steps involved in properly deleting items: see https://github.com/ElDewrito/ElDorito/blob/master/ElDorito/Source/Blam/BlamData.cpp#L18.

I would prefer that you refactor this file to use a data array for content items and then follow the steps for properly deleting elements. Consider implementing a `DataArrayBase::Clear()` method.